### PR TITLE
Implement dependency-based (partial) ordering for parallel initialization tasks

### PR DESCRIPTION
**Pull request by Technici4n**
This just adds a test for https://github.com/neoforged/FancyModLoader/pull/123, and bumps our FML dependency.

### CODE REVIEW

File: tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    private static void onConstruct(int index) {
+        synchronized (lock) {
+            if (constructedMod != index - 1) {
+                throw new IllegalStateException("Mod " + constructedMod + " was constructed, cannot construct " + index);
+            }
+        }
+
+        try {
+            Thread.sleep(10);
```
Comment (embeddedt) [inline tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java:22]: Does the delay serve any purpose?

Comment (Technici4n) [reply tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java:22]: Yes, it ensures that we can detect with high probability any attempt at scheduling mod N+1 before mod N is finished completing. FYI these are milliseconds not seconds.

Comment (embeddedt) [reply tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java:22]: That feels like a pretty flaky approach to me, but fine.

**Review by embeddedt — APPROVED** [review]

File: tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java (hunk 1 above)
Comment (Technici4n) [reply tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java:22]: What else would you suggest? Multi-threading is inherently non-deterministic. I will note that this code managed to crash when the sorting logic was not implemented correctly in FML every single time, so at least it works. :P

Comment (embeddedt) [reply tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java:22]: I would have made the timeout longer (maybe 1 second) to minimize flakiness from the OS being busy, but then you would have to reduce the number of mods to make the test not take too long to run.

Comment (Technici4n) [reply tests/src/main/java/net/neoforged/neoforge/oldtest/OrderedModLoadingTest.java:22]: I feel like 5 is already not that much. 😅 Well this is probably fine as-is anyway.

**Pull request merged**